plugins/node/opentelemetry-instrumentation-nestjs-core/test/index.test.ts

```diff
@@ -0,0 +1,222 @@
+      { service: 'test', name: 'Create Nest App', module: 'AppModule' },
+    ]);
+  });
+
+  it('should capture requests', async () => {
+    const path = semver.intersects(LIB_VERSION, '<5.0.0') ? '/' : '/users';
+    const url = '/users';
+    const instance = 'UsersController';
+    const callback = 'getUsers';
+
+    assert.strictEqual(await request('/users'), 'Hello, world!\n');
+
+    assertSpans(memoryExporter.getFinishedSpans(), [
+      { service: 'test', name: 'Create Nest App', module: 'AppModule' },
+      {
+        service: 'test',
+        name: callback,
+        callback,
+        parentSpanName: `${instance}.${callback}`,
+      },
+      {
+        service: 'test',
+        name: `${instance}.${callback}`,
+        method: 'GET',
+        url,
+        path,
+        callback,
+      },
+    ]);
+  });
+
+  it('should capture errors', async () => {
+    const path = semver.intersects(LIB_VERSION, '<5.0.0') ? '/' : '/errors';
+    const url = '/errors';
+    const instance = 'ErrorController';
+    const callback = 'getError';
+
+    assert.strictEqual(
+      await request('/errors'),
+      '{"statusCode":500,"message":"Internal server error"}'
+    );
+
+    assertSpans(memoryExporter.getFinishedSpans(), [
+      { service: 'test', name: 'Create Nest App', module: 'AppModule' },
+      {
+        service: 'test',
+        name: callback,
+        callback,
+        status: {
+          code: SpanStatusCode.ERROR,
+          message: 'custom error',
+        },
+        parentSpanName: `${instance}.${callback}`,
+      },
+      {
+        service: 'test',
+        name: `${instance}.${callback}`,
+        method: 'GET',
+        url,
+        path,
+        callback,
+        status: {
+          code: SpanStatusCode.ERROR,
+          message: 'custom error',
+        },
+      },
+    ]);
+  });
+});
+
+const assertSpans = (actualSpans: any[], expectedSpans: any[]) => {
+  assert(Array.isArray(actualSpans), 'Expected `actualSpans` to be an array');
+  assert(
+    Array.isArray(expectedSpans),
+    'Expected `expectedSpans` to be an array'
+  );
+  assert.strictEqual(
+    actualSpans.length,
+    expectedSpans.length,
+    'Expected span count different from actual'
+  );
+
+  actualSpans.forEach((span, idx) => {
+    const expected = expectedSpans[idx];
+    if (expected === null) return;
+    try {
+      assert.notStrictEqual(span, undefined);
+      assert.notStrictEqual(expected, undefined);
+
+      assert.strictEqual(span.attributes.component, '@nestjs/core');
+      assert.strictEqual(span.attributes['nestjs.module'], expected.module);
+
+      assert.strictEqual(span.name, expected.name);
+
+      assert.strictEqual(span.attributes['http.method'], expected.method);
+      assert.strictEqual(span.attributes['http.url'], expected.url);
+      assert.strictEqual(span.attributes['http.route'], expected.path);
+      assert.strictEqual(span.attributes['nestjs.callback'], expected.callback);
+      assert.strictEqual(
```

Comment:
I don't see any asserts covering `nestjs.type` - it looks your tests cover all the types of spans your instrumentation creates, but I don't see anything validating this attribute?

Reply:
That is right! Thanks!